### PR TITLE
TST: groupby-reindex on DTI

### LINKED ISSUE #26209: reindex does not work for groupby series with DateTimeIndex
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np

# generate a simple df with weekly DateTimeIndex, group and a value
df = pd.DataFrame({
    'group':['Group1','Group2','Group3']*3, 
    'value':np.random.randint(100, 1000, size=9)}, 
    index=pd.date_range('1991-10-2',periods=3, freq='W-MON').repeat(3)
)

# this works as expected
new_value = df.groupby('group')[['value']].apply(
    lambda x: x.reindex(pd.date_range(x.index.min(), x.index.max(), freq='W-MON'), fill_value=0)
)

# this fails
new_value = df.groupby('group').value.apply(
    lambda x: x.reindex(pd.date_range(x.index.min(), x.index.max(), freq='W-MON'), fill_value=0)
)
```
#### Problem description

We have a DF with a group and DateTimeIndex, we want to `reindex` the values per group to make sure all groups have all the appropriate weekly DateTimeIndex. This works as expected when there is a gap to be resolved by the `reindex`ing... However, it fails when there's no gap to be filled by `reindex`.

`ValueError: cannot reindex from a duplicate axis`

This issue is resolved by turning the corresponding Series into a DF by `[[]]` syntax.

#### Expected Output

The original Series.

#### Output of ``pd.show_versions()``

<details>
installed versions
------------------
commit: none
python: 3.6.8.final.0
python-bits: 64
os: linux
os-release: 4.4.0-43-microsoft
machine: x86_64
processor: x86_64
byteorder: little
lc_all: none
lang: en_us.utf-8
locale: en_us.utf-8

pandas: 0.24.2
pytest: 3.5.1
pip: 19.0.3
setuptools: 41.0.0
cython: 0.28.2
numpy: 1.14.3
scipy: 1.1.0
pyarrow: none
xarray: none
ipython: 6.4.0
sphinx: 1.7.4
patsy: 0.5.0
dateutil: 2.8.0
pytz: 2019.1
blosc: none
bottleneck: 1.2.1
tables: 3.4.3
numexpr: 2.6.5
feather: none
matplotlib: 2.2.2
openpyxl: 2.5.3
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.4
lxml.etree: 4.2.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.7
pymysql: none
psycopg2: none
jinja2: 2.10
s3fs: none
fastparquet: none
pandas_gbq: none
pandas_datareader: none
gcsfs: None
</details>

### PR DESCRIPTION
**Pull request by CloseChoice**
Add test to check whether reindexing works correctly.

- [ ] closes #26209
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
I think need to run black on the change to get CI to pass

**Comment by CloseChoice** [comment]
> I think need to run black on the change to get CI to pass

done and now CI-linting passes

File: pandas/tests/groupby/test_apply.py [hunk 1]
```diff
@@ -901,3 +901,17 @@ def fn(x):
...
+    # GH: 26209
+    values = [1, 2, 3, 4]
+    indices = [1, 1, 2, 2]
+    df = pd.DataFrame(
+        {"group": ["Group1", "Group2"] * 2, "value": values}, index=indices
+    )
+    srs_exp = pd.Series(values, index=indices, name="value")
+    srs_grouped = df.groupby("group").value.apply(
+        lambda x: x.reindex(np.arange(x.index.min(), x.index.max() + 1))
+    )
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_apply.py:916]: theres a lot going on here.  can you break this up into multiple lines and comment on which one used to be a problem

File: pandas/tests/groupby/test_apply.py [hunk 2]
```diff
@@ -901,3 +901,17 @@ def fn(x):
...
+    tm.assert_series_equal(srs_exp, srs_grouped)
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_apply.py:917]: can you call these `expected` and `result`

File: pandas/tests/groupby/test_apply.py [hunk 3]
```diff
@@ -901,3 +901,17 @@ def fn(x):
         name="col2",
     )
     tm.assert_series_equal(result, expected)
+
+
+def test_apply_reindex_values():
+    # GH: 26209
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_apply.py:907]: a few words about what the bug was would be hepful

Comment (CloseChoice) [reply pandas/tests/groupby/test_apply.py]: done

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @CloseChoice

**Pull request merged**